orchestra/todos/views.py

```diff
@@ -78,6 +86,44 @@ def worker_task_recent_todo_qas(request):
     return Response(todos_recommendation)
 
 
+@method_decorator(staff_member_required, name='dispatch')
+class ImportTodoListTemplateFromSpreadsheet(View):
+    TEMPLATE = 'orchestra/import_todo_list_template_from_spreadsheet.html'
+
+    def get(self, request, pk):
+        # Display a form with a spreadsheet URL for import.
+        return render(
+            request,
+            self.TEMPLATE,
+            {'form': ImportTodoListTemplateFromSpreadsheetForm(initial={})})
+
+    def post(self, request, pk):
+        # Try to import the spreadsheet and redirect back to the admin
+        # entry for the imported TodoListTemplate. If we encounter an
+        # error, display the error on the form.
+        form = ImportTodoListTemplateFromSpreadsheetForm(request.POST)
+        context = {'form': form}
+        if form.is_valid():
+            try:
+                todo_list_template = TodoListTemplate.objects.get(
+                    id=pk)
+                import_from_spreadsheet(
+                    todo_list_template,
+                    form.cleaned_data['spreadsheet_url'],
+                    request)
+                return redirect(
+                    'admin:orchestra_todolisttemplate_change',
```

Comment:
is there a way to add an 'import successful' badge to the top of the change admin view when you redirect? I vaguely recall that being possible.

Reply:
ask and you shall receive

<img width="1204" alt="image" src="https://user-images.githubusercontent.com/57725/93260184-0412e880-f76f-11ea-8b86-deecb27ce7d2.png">